### PR TITLE
58745 VAMC updates

### LINKED ISSUE #58745: Bug Fix - Changes for VAMC field in Supporting Evidence
## Issue Description
The scope of this ticket is to make the following changes around the VAMC field in Supporting Evidence page:
- The VAMC entry should be optional on this page (currently required)
- Updated content to clearly explain what is needed in this field
- The field should not force validation and require specific/accurate names of the facility
- Country field is optional and remove validation (currently required)
- Condition field is optional and remove validation (currently required)
- All fields are optional on this page
- User should be able to submit claim with no value or unvalidated value for these fields

## Need to know
Look at backend validation on this: https://github.com/department-of-veterans-affairs/vets-json-schema/blob/76083e33f175fb00392e31f1f5f90654d05f1fd2/dist/21-526EZ-ALLCLAIMS-schema.json#L1016

https://www.sketch.com/s/5e464026-9acb-4f80-a4f9-634c7e972d19

This is based on our discussion with VBA stakeholders(Alejandro) on [May 18th](https://docs.google.com/document/d/1H9xgYJ-oAj3AWKtnzZxOI3xv_ayQKju03rkc36Emax8/edit#)
Remove requirement of entering VAMC name on the evidence screen
- Multiple issues are being reported from veterans with their inability to choose VAMC from options or enter a name and move to the next step or submit the form
- Even when the value is entered, it may be incorrect or incomplete, leading more work by VSR to check all records
- We understand that regardless of a veteran entering this value, VSRs look at all VAMCs in the system to assess if it’s relevant to the claim
We propose to either remove this field altogether or at least making it optional so the user does not see this issue.
 - Don’t validate the facilities name - VSR will do the research  
- Make it optional
- Clarify the language to what is required for this field

## Update 6/15/23
We found out that these newly optional fields are required in EVSS ([slack thread](https://dsva.slack.com/archives/C1VBAHWQL/p1686785454385369)) For now, we will move forward with only the content changes and removing auto suggest from the facilities name field. That should hopefully allow users to proceed. 

AC has been updated accordingly and we will file a subsequent story to be worked on next month, once the 3 fields are optional in EVSS.

---
## Tasks
- [x] Update automated tests
- [ ] ~~Backend tests~~
- [x] End to end tests

## Acceptance Criteria
- [x] Content updates to remove and edit text as noted in the sketch file
- [x] User will be able to enter a facility name that does not match any facility in the VAMC list
- [ ] ~~User will be able to enter a Country, Condition without validation~~
- [ ] ~~User will be able to submit the claim without entering any value for VAMC, Country, Condition~~
- [x] Verify no harmful downstream effects

---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `Console-Services`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

### PR DESCRIPTION
**Pull request by christinec-fftc**
## Summary
-Remove text and update text on the va-medical-records page
-Make the Name of VA medical facility input field free form instead of autosuggest from pre-defined list
-Cleanup unused code

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/58745

## Testing done
- _Describe what the old behavior was prior to the change_
Prior to this change the facility name was using auto suggest, but users would get stuck here since the api is not working

- _Describe the steps required to verify your changes are working as expected_
1. Start fllling out an original claim
2. When you get to the supporting-evidence/evidence-types page, select 'VA medical records' for evidence types. Continue to the next page
3. Verify you can type in a facility name, other required fields and then save or continue
Note : the behavior of the array list component is to do a scroll up when clicking the save button (it will not collapse until you click 'Add another facility' button)
4. Complete the rest of the 526 form and verify you can submit successfully

- _Describe the tests completed and the results_
Manual: created and submitted (original) claims with both new and rated conditions
Unit: minor updates for unit tests
e2e tests: verified all tests still working

## Screenshots
image
image

## What areas of the site does it impact?

526EZ form on the va-medical-records page

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/tests/pages/vaMedicalRecords.unit.spec.jsx [hunk 1]
```diff
@@ -188,7 +197,9 @@ describe('VA Medical Records', () => {
     form.unmount();
   });
 
-  it('should submit with all required info', () => {
+  it('should submit with sparse facility info', () => {
+    // Treatment facility is missing a name, date, and city (they are optional). VSRs look at all VAMCs in the
+    // system to assess if it’s relevant to the claim.
```
Comment (mchae-nava) [inline src/applications/disability-benefits/all-claims/tests/pages/vaMedicalRecords.unit.spec.jsx:202]: Appreciate the clarifying comment. This is pretty neat, and I'm curious how this check works. Definitely makes things so much more convenient for veterans

**Review by mchae-nava — APPROVED** [review]
LGTM!

File: src/applications/disability-benefits/all-claims/pages/vaMedicalRecords.js [hunk 2]
```diff
@@ -45,17 +42,9 @@ export const uiSchema = {
...
-          'ui:options': { queryForResults: true, freeInput: true },
-          'ui:errorMessages': {
-            maxLength: 'Please enter a name with fewer than 100 characters.',
-            pattern: 'Please enter a valid name.',
-          },
-        },
-      ),
+      treatmentCenterName: {
+        'ui:title': 'Name of VA medical facility',
+      },
```
[the remark this answers is not in the document]
Comment (mchae-nava) [reply src/applications/disability-benefits/all-claims/pages/vaMedicalRecords.js:47]: Ah whoops, forgot to delete this. The tests you wrote answered my question

[the remark this answers is not in the document]
Comment (christinec-fftc) [reply src/applications/disability-benefits/all-claims/pages/vaMedicalRecords.js:46]: That's correct. The api was removed a while back (for reference https://github.com/department-of-veterans-affairs/vets-website/pull/16016 and https://github.com/department-of-veterans-affairs/va.gov-team/issues/57981)

This should be a free form field now without auto suggest.

**Review by sethdarragile6 — APPROVED** [review]

**Pull request merged**